src/components/checkbox/checkbox.tsx

```diff
@@ -38,32 +39,65 @@ const HiddenCheckbox = styled.input.attrs({ type: "checkbox" })`
   width: 1px;
 `;
 
-const StyledCheckbox = styled.div<{ checked: boolean }>`
+const StyledCheckbox = styled.div<{ checkedState: SelectionState }>`
   width: 24px;
   height: 24px;
-  background: ${({ checked }) =>
-    checked ? color_brand_tertiary : "transparent"};
+  background: ${({ checkedState }) => {
+    switch (checkedState) {
+      case SelectionState.selected:
+      case SelectionState.partiallySelected:
+        return color_brand_tertiary;
+      default:
+        return "transparent";
+    }
+  }};
   border: 1px solid ${color_brand_tertiary};
   border-radius: 4px;
 
   &:hover {
-    background: ${({ checked }) =>
-      checked ? dim_brand_tertinary : "transparent"};
+    background: ${({ checkedState }) => {
+    switch (checkedState) {
+      case SelectionState.selected:
+      case SelectionState.partiallySelected:
+        return dim_brand_tertinary;
+      default:
+        return "transparent";
+    }
+  }}
     border: 1px solid ${dim_brand_tertinary};
   }
 
   ${Icon} {
-    visibility: ${(props) => (props.checked ? "visible" : "hidden")};
-  }
+    visibility: ${({ checkedState }) => {
+    switch (checkedState) {
+      case SelectionState.selected:
+      case SelectionState.partiallySelected:
+        return "visible";
+      default:
+        return "hidden";
+    }
+  }};
 `;
 
-export const Checkbox = ({ id, checked, onChange, ...rest }: CheckboxProps) => (
-  <CheckboxContainer>
-    <HiddenCheckbox id={id} checked={checked} onChange={onChange} {...rest} />
-    <StyledCheckbox checked={checked}>
-      <Icon id={`${id}-icon`} viewBox="0 0 24 24">
-        <polyline points="20 6 9 17 4 12" />
-      </Icon>
-    </StyledCheckbox>
-  </CheckboxContainer>
-);
+export const Checkbox = ({ id, onChange, checked: checkedState, ...rest }: CheckboxProps) => {
+  const checked = checkedState == SelectionState.selected;
+
+  return (
+    <CheckboxContainer>
+      <HiddenCheckbox id={id} checked={checked} onChange={onChange} {...rest} />
+      <StyledCheckbox id={`${id}-icon`} checkedState={checkedState}>
+        {checkedState === SelectionState.selected &&
+          (<Icon  viewBox="0 0 24 24">
+            <polyline points="20 6 9 17 4 12" />
+          </Icon>)
+        }
+        {checkedState === SelectionState.partiallySelected &&
+          (<Icon width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
+            <rect x="6" y="12" width="12" height="1" />
+          </Icon>)
```

Comment:
Could you move the `svg` to a separate file? It looks like we used it this way before we started to use `svgr`

Reply:
Done